File: tests/fixtures/code/python/sample.py
# SPDX-FileCopyrightText: 2026 The Nikasha Authors
# SPDX-License-Identifier: Apache-2.0
import json


def load(path):
    with open(path) as fh:
        return json.load(fh)


class Store:
    """A tiny key-value store."""

    def __init__(self, path):
        self.data = load(path)

    @property
    def size(self):
        return len(self.data)

    async def fetch(self, key):
        def pick(d):
            return d.get(key)

        return pick(self.data)

    class Meta:
        def describe(self):
            return repr(self)


def main():
    store = Store("db.json")
    print(store.size)
    return store.fetch("k")


main()

File: tests/fixtures/code/ruby/sample.rb
# SPDX-FileCopyrightText: 2026 The Nikasha Authors
# SPDX-License-Identifier: Apache-2.0
require 'json'

module Billing
  class Invoice
    attr_reader :total

    def initialize(lines)
      @total = sum(lines)
    end

    def self.parse(text)
      new(JSON.parse(text))
    end

    def to_s
      format('%.2f', total)
    end

    private

    def sum(lines)
      lines.map { |l| l.fetch(:amount) }.sum
    end
  end

  def self.version
    '1.0'
  end
end

def helper(x)
  puts Billing::Invoice.parse(x)
end
